src/material/slider/slider.ts

```diff
@@ -625,705 +152,757 @@ export class MatSlider
-            this._layout();
-            this._cachedHostRect = null;
-          }
-        }, 50);
-      });
-      this._resizeObserver.observe(this._elementRef.nativeElement);
-    });
+  /** Whether or not the slider thumbs overlap. */
+  private _thumbsOverlap: boolean = false;
+
+  /** Returns true if the slider knobs are overlapping one another. */
+  private _areThumbsOverlapping(): boolean {
+    const startInput = this._getInput(Thumb.START);
+    const endInput = this._getInput(Thumb.END);
+    if (!startInput || !endInput) {
+      return false;
+    }
+    return endInput.translateX - startInput.translateX < 20;
   }
 
-  /** Whether any of the thumbs are currently active. */
-  private _isActive(): boolean {
-    return this._getThumb(Thumb.START)._isActive || this._getThumb(Thumb.END)._isActive;
+  /**
+   * Updates the class names of overlapping slider thumbs so
+   * that the current active thumb is styled to be on "top".
+   */
+  private _updateOverlappingThumbClassNames(source: MatSliderRangeThumbInterface): void {
+    const sibling = source.getSibling()!;
+    const sourceThumb = this._getThumb(source.thumbPosition);
+    const siblingThumb = this._getThumb(sibling.thumbPosition);
+    siblingThumb._hostElement.classList.remove('mdc-slider__thumb--top');
+    this._thumbsOverlap
+      ? sourceThumb._hostElement.classList.add('mdc-slider__thumb--top')
+      : sourceThumb._hostElement.classList.remove('mdc-slider__thumb--top');
+  }
+
+  /** Updates the UI of slider thumbs when they begin or stop overlapping. */
+  private _updateOverlappingThumbUI(source: MatSliderRangeThumbInterface): void {
+    if (!this._isRange || this._skipUpdate()) {
+      return;
+    }
+    if (this._thumbsOverlap !== this._areThumbsOverlapping()) {
+      this._thumbsOverlap = !this._thumbsOverlap;
+      this._updateOverlappingThumbClassNames(source);
+    }
   }
-}
 
-/** The MDCSliderAdapter implementation. */
-class SliderAdapter implements MDCSliderAdapter {
-  /** The global event listener subscription used to handle events on the slider inputs. */
-  private _globalEventSubscriptions = new Subscription();
+  // Thumb styles update conditions
+  //
+  // 1. TranslateX, resize, or dir change
+  //    - Reason: The thumb styles need to be updated according to the new translateX.
+  // 2. Min, max, or step
+  //    - Reason: The value may have silently changed.
+
+  /** Updates the translateX of the given thumb. */
+  _updateThumbUI(source: MatSliderThumbInterface) {
+    if (this._skipUpdate()) {
+      return;
+    }
 
-  /** The MDC Foundations handler function for start input change events. */
-  private _startInputChangeEventHandler: SpecificEventListener<EventType>;
+    const transform = `translateX(${source.translateX}px)`;
+    source.thumbPosition === Thumb.END
+      ? (this._getThumb(Thumb.END)!._hostElement.style.transform = transform)
+      : (this._getThumb(Thumb.START)!._hostElement.style.transform = transform);
```

Comment:
Should be able to simplify to the following to avoid some of the repetition: 

```
const thumb = this._getThumb(source.thumbPosition === Thumb.END ? Thumb.END : Thumb.START)!;
thumb._hostElement.style.transform = `translateX(${source.translateX}px)`;
```